Build - make misc updates for having precompiled tau assets in build

lilyinstarlight opened the pull request:
@samaaron for using precompiled assets instead of running esbuild/tailwind/etc during build, do you just want the results from `mix assets.deploy.prod` committed to the repo?

If so we can remove those files from the clean scripts, gitignore, etc and just let them get committed. This PR does that plus make a few other misc changes to keep things consistent after e907632 and 5b8bc5b

Specific changes in this PR:
* Ignore precompiled assets in clean scripts
* Remove precompiled assets from gitignore
* Run `assets.deploy.prod` mix command when doing pre-tau-dev-init
* Remove `assets.deploy.prod` command from linux offline build and add `deps.compile`
* Respect `MIX_DEPS_PATH` from tailwind config
* Remove reliance on tailwind/esbuild dependencies in test env

samaaron commented:
Yep, this looks very much like the direction I was heading in. This stuff is still very much in an embryonic form and It's not even featuring in v4. However, the tech is all there just to make sure that the foundations are there in a form that can serve existing needs and to be ready for activity post v4's release.

I'm sure there will be much more manipulation of this kind of stuff going forwards as we start building actual user-facing functionality :-)

lilyinstarlight commented:
Alright, if this is what you had in mind, I'll mark this ready for review after I push a few more changes. I noticed CI started timing out after 5b8bc5b for macOS and Linux, so I may try to investigate that if I have time today. It might be related to not having precompiled files it expects to exist when starting Tau (since Tau is what is failing to start judging by logs, which causes CI to hang like this)

lilyinstarlight commented:
Okay, I think I've fixed the CI issue. The tailwind and esbuild tasks were unavailable for `mix assets.deploy.dev` when `MIX_ENV=test` because of the `only: :dev` specifier, so I updated those to use `only: [:dev, :test]`
Edit: Alternatively we can just remove the `mix assets.deploy.dev` command from the Tau test env in boot-{lin,mac,win} scripts since we don't expect them to be generated at runtime anymore. After thinking about it that actually seems like a more sensible solution

lilyinstarlight committed:
Build - make misc updates for having precompiled tau assets in build

Updates:
* ignore precompiled assets in clean scripts
* remove precompiled assets in gitignore
* run `assets.deploy.prod` mix command when doing pre-tau-dev-init
* remove `assets.deploy.prod` command from linux offline build and add
  `deps.compile`
* respect MIX_DEPS_PATH from tailwind config
* remove reliance on tailwind/esbuild in test env

samaaron commented:
Wonderful, thank-you so much for this :-)
